blosc/frame.c

```diff
@@ -551,11 +570,21 @@ int frame_update_trailer(blosc2_frame* frame, blosc2_schunk* schunk) {
     memcpy(frame->sdata + trailer_offset, trailer, trailer_len);
   }
   else {
-    FILE* fp = fopen(frame->fname, "rb+");
+    FILE* fp = NULL;
+    if (frame->eframe) {
+      char* eframe_name = malloc(strlen(frame->urlpath) + strlen("/chunks.b2frame") + 1);
+      sprintf(eframe_name, "%s/chunks.b2frame", frame->urlpath);
+      fp = fopen(eframe_name, "rb+");
+      free(eframe_name);
+    }
+    else {
+      fp = fopen(frame->urlpath, "rb+");
+    }
     fseek(fp, trailer_offset, SEEK_SET);
     size_t wbytes = fwrite(trailer, 1, trailer_len, fp);
     if (wbytes != (size_t)trailer_len) {
       BLOSC_TRACE_ERROR("Cannot write the trailer length in trailer.");
+      fclose(fp);
```

Comment:
Consider moving this `fclose(fp)` and the one below it above the `if` statement.

Reply:
Fixed in 4607ab374e512f47bc7a64117e9af8907c18fc81